components/places/src/storage/bookmarks/root_guid.rs

```diff
@@ -89,26 +91,26 @@ impl From<BookmarkRootGuid> for SyncGuid {
 // Allow comparisons between BookmarkRootGuid and SyncGuids
 impl PartialEq<BookmarkRootGuid> for SyncGuid {
     fn eq(&self, other: &BookmarkRootGuid) -> bool {
-        self.0 == other.as_str()
+        self.as_str() == other.as_str()
```

Comment:
micronit: this would avoid an unnecessary from_utf8 which we'll be causing when removing the unsafe.
```suggestion
        self.as_bytes() == other.as_str().as_bytes()
```

Please make similar changes for the guid tests below as well.